service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendEhrExtractCore.java

```diff
@@ -0,0 +1,33 @@
+package uk.nhs.adaptors.gp2gp.ehr;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.task.TaskDispatcher;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class SendEhrExtractCore {
```

Comment:
This class name confused me. The name implies that is sends the EHR Extract "Core" message to MHS.

Reply:
Could it be SendEhrExtractCoreTaskService with method send ? SendEhrExtractCore is a task type that will send EHR Extract to Spine

Reply:
Maybe `SendEhrExtractCoreTaskDispatcher`? its a mouthful though! Maybe someone else will have a better name...

Reply:
I thought about TaskDispatcher as well, it is ok with me as well, the class is responsible for creating/sending new tasks.